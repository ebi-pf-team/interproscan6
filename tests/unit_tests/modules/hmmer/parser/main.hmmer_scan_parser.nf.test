nextflow_process {

    name "Test Process HMMER_SCAN_PARSER"
    script "interproscan/modules/hmmer/parser/main.nf"
    process "HMMER_SCAN_PARSER"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmscan_parser/hmmscan.out"
        symlink "interproscan/scripts/hmmer/parser_scan_out.py"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmscan_parser/hmmscan.out")
                input[1] = "PIRSF"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
