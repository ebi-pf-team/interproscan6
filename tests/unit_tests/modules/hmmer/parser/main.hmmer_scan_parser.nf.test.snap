{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hmmer_parsed_hmmscan.out.json:md5,0ef26ce77584e95f4ab3b3a514591b8f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T08:57:03.130917"
    }
}